kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/InstallKT.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assumptions.assumeThat;
+import static org.assertj.core.api.Assumptions.assumeThatCode;
+
+class InstallKT {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(InstallKT.class);
+
+    public static final String SYS_PROP_IMAGE_ARCHIVE = "io.kroxylicious.operator.image.archive";
+    public static final String SYS_PROP_IMAGE_NAME = "io.kroxylicious.operator.image.name";
+    // These are normally set automatically by mvn
+    private static final String IMAGE_ARCHIVE = System.getProperty(SYS_PROP_IMAGE_ARCHIVE);
+    private static final String IMAGE_NAME = System.getProperty(SYS_PROP_IMAGE_NAME);
+    private static boolean loaded = false;
+
+    @BeforeAll
+    static void beforeAll() throws IOException, InterruptedException {
+        Assertions.setDescriptionConsumer(desc -> {
+            LOGGER.info("{}", desc);
+        });
+        assumeThat(IMAGE_ARCHIVE)
+                .describedAs("System property %s must be set", SYS_PROP_IMAGE_ARCHIVE)
+                .isNotNull();
+        assumeThat(Path.of(IMAGE_ARCHIVE))
+                .describedAs("Container image archive %s must exist", IMAGE_ARCHIVE)
+                .exists();
+
+        assumeThat(IMAGE_NAME)
+                .describedAs("System property %s must be set", SYS_PROP_IMAGE_NAME)
+                .isNotNull();
+
+        LOGGER.info("Checking whether minikube is available");
+        assumeThatCode(() -> exec("minikube"))
+                .describedAs("minikube must be available on the path")
+                .doesNotThrowAnyException();
+
+        LOGGER.info("Checking whether minikube is running");
+        assumeThatCode(() -> exec("minikube", "status"))
+                .describedAs("minikube must be running")
+                .doesNotThrowAnyException();
```

Comment:
I'm not a big fan of asserting minikube is available and running as I tend to use CRC running on a linux box instead of a VM locally on my mac. Currently the test only relies on having kubetl logged into a cluster maybe we can find a compromise around that.

Reply:
I think that's something for a follow-on PR, but I agree it's worth supporting. 

If we pursued #2112 I would rewrite this test as a `proc` with a notional prereq which could be satisfied using minikube or CRC. Or if not I think there's really an abstract test here with two concrete classes which provide their own `BeforeAll`/`AfterAll`.

Reply:
Yeah I'm happy with that. I've no desire to stall things for my odd ball workflow.